packages/webapp/src/routes/land/add-planning-authority.js

```diff
@@ -39,7 +42,21 @@ const handlers = {
         localPlanningAuthorityNameErr
       })
     }
-    const lpaList = request.yar.get(constants.redisKeys.PLANNING_AUTHORTITY_LIST) ?? []
+    const excludeIndex = id !== undefined ? parseInt(id, 10) : null
+    const duplicateError = checkForDuplicate(
+      lpaList,
+      null,
+      selectedLpa,
+      '#localPlanningAuthority',
+      'This local planning authority has already been added - enter a different local planning authority, if there is one',
+      excludeIndex
+    )
+    if (duplicateError) {
+      return h.view(constants.views.ADD_PLANNING_AUTHORITY, {
+        err: Object.values(duplicateError),
+        localPlanningAuthorityNameErr
+      })
+    }
```

Comment:
if lpaList.includes(selectedLpa) then its duplicate right? I guess this not cover all scenarios?

Reply:
as per discussion excludeIndex required for edit functionality